index.d.ts

```diff
@@ -32,6 +59,23 @@ declare namespace cpy {
 		```
 		*/
 		readonly rename?: string | ((basename: string) => string);
+
+		/**
+		Function to filter files to copy. Should accept source file object as argument.
+		Return true to include, false to exclude. Can also return a Promise that resolves to true or false.
+
+		@example
+		```
+		import cpy = require('cpy');
+
+		(async () => {
+			await cpy('foo', 'destination', {
+				filter: file => file.extension !== '.nocopy'
```

Comment:
`.extension` should not include the `.`

Reply:
Shouldn't it? Native `path` parse(...) method returns it with `.`, so I fugured it would be good to mirror this choice.

Reply:
If it shouldn't, then what should we do with multiple dots, remove them all?

As in `abc..gz` => `.gz` or `gz`? (`..gz` atm)

Reply:
I don't really care what the Node.js methods do. They're usually not very good nor user-friendly.

Reply:
> If it shouldn't, then what should we do with multiple dots, remove them all?

I've never seen multiple consecutive dots, but you should preserve it on the name. `abc..gz` => (`abc.` and `gz`).